plugin/metadata/README.md

```diff
@@ -0,0 +1,39 @@
+# metadata
+
+## Name
+
+*metadata* - enable  a metadata collector.
+
+## Description
+
+By enabling *metadata* any plugin that implements [metadata.metadater interface](https://godoc.org/github.com/coredns/coredns/plugin/metadata#Metadater) will be called and will be able to add it's own Metadata to context. The metadata collected will be available for all plugins handler, via the Context parameter provided in the ServeDNS function
+
```

Comment:
This plugin needs to do a few other things, too:

1) It needs to manage the "built-in" variables that you would find in `rewrite`. That is, when executed, it should create metadata entries for all those built-in variables. Note that it should still be possible to use `rewrite` *without* `metadata`, so you may want to refactor some stuff to enable some code sharing there.

2) It needs to enable the parsing of EDNS0 options in the same way that the [Themis policy plugin](https://github.com/infobloxopen/themis/tree/master/contrib/coredns/policy) does. This can come in a later PR, but put it in your plans.

The README should explain some use cases - I can add them later if you want.

Reply:
Regarding part 1. Could you clarify the idea? I see at least 3 ways which may fit:
1. Rewrite plugin implements Metadataer to add them to metadata in case if other plugins need them. Other logic remains the same (support of additional variables by rewrite is added in any case).
2. "build-in" variables are moved to metadata plugin which implements Metadataer to add them to metadata. Rewrite plugin will remain functional, but "built-in" variables are not available if metadata is disabled.
3. Same as 2, but variables are supposed to be accessible somehow with no metadata plugin enabled. I don't see how this can be implemented with no conflict with conception of plugins.

Reply:
@johnbelamaric : can you reply to Eugen as soon as you can so he can go ahead ? Thanks!